### PR TITLE
feat(rust): Rust bindings for the cycles ledger

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
The Oisy backend canister will soon have to make API calls to the cycles ledger.

# Changes
- Add Rust bindings for calling the cycles ledger.

# Tests
This PR defines type signatures, so the relevant test is `cargo check` passing.

# TODO
The bindings were generated using `didc` and then derive traits such as `Eq` and `PartialEq` were added to particular types that will need them.  This derivation can probably be automated with the shiny new `didc` configuration options.  [Tracking issue.](https://dfinity.atlassian.net/browse/GIX-3032)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,6 +10,8 @@ resolver = "2"
 ic-cdk = "0.15.1"
 ic-cdk-macros = "0.15.0"
 ic-cdk-timers = "0.9.0"
+ic-cycles-ledger-client = { path = "src/cycles_ledger_client" }
+ic-ledger-types = "0.13.0"
```
Comment (lmuntaner) [inline Cargo.toml:14]: Where is this used? Of will it be used in an upcoming PR?

File: src/cycles_ledger_client/src/lib.rs
Comment (lmuntaner) [inline src/cycles_ledger_client/src/lib.rs:1]: Is this file created with a script or manually?

Can we add which script was used if so?

**Review by lmuntaner — APPROVED** [review]
A couple of comments, but already approving.

File: Cargo.toml (hunk 1 above)
Comment (bitdivine) [reply Cargo.toml:14]: It will be used in an upcoming PR.

File: src/cycles_ledger_client/src/lib.rs
Comment (bitdivine) [reply src/cycles_ledger_client/src/lib.rs:1]: Manually.  There is a TODO section in the MR description with details.

Comment (lmuntaner) [reply src/cycles_ledger_client/src/lib.rs:1]: Thanks!

**Pull request merged**